File: Math/Easy/70/70.py
# class Solution:
#     def climbStairs(self, n: int) -> int:
#         arr = [0] * (n)
#         for i in range(n):
#             if i < 3:
#                 arr[i] = i + 1
#             else:
#                 arr[i] = arr[i-1] + arr[i-2]
#         return arr[-1]

class Solution:
    def climbStairs(self, n: int) -> int:
        x, y = 1, 1
        for i in range(n-1):
            z = x + y
            x = y
            y = z
        return y

File: Math/Easy/70/70.c
#include <stdio.h>
#include <stdlib.h>

// int climbStairs(int n){
//     int i = 1, j = 1;
//     int k = 0;
//     for(int x = 0; x < n - 1; x++){
//         k = i + j;
//         i = j;
//         j = k;
//     }
//     return j;
// }

int arr[100];

int climbStairs(int n){
    if(n < 3){
        return n;
    }
    if(arr[n] == NULL){
        arr[n] = climbStairs(n - 1) + climbStairs(n + 2);
    }
    return arr[n];
}